src/packet.ml

```diff
@@ -385,18 +396,26 @@ let decode_tls_data ?(with_premaster = false) buf =
      let p_len = Cstruct.BE.get_uint16 buf p_start in
      guard (Cstruct.length buf >= p_start + 2 + u_len + p_len) `Partial
      >>= fun () ->
-     maybe_string "password" buf (p_start + 2) p_len >>| fun p ->
-     let end_of_data = p_start + 2 + p_len in
-     (* for some reason there may be some slack here... *)
-     if Cstruct.length buf > end_of_data then
-       let data = Cstruct.shift buf end_of_data in
-       Log.warn (fun m ->
-           m "%u bytes slack at end of tls_data %s (p is %s)@.%a"
-             (Cstruct.length data) (Cstruct.to_string data) p Cstruct.hexdump_pp
-             data)
-     else ();
-     match (u, p) with "", "" -> None | _ -> Some (u, p))
-  >>| fun user_pass -> { pre_master; random1; random2; options; user_pass }
+     maybe_string "password" buf (p_start + 2) p_len >>= fun p ->
+     let user_pass = match (u, p) with "", "" -> None | _ -> Some (u, p) in
+     let peer_info_start = p_start + 2 + p_len in
+     (if Cstruct.length buf = peer_info_start then Ok None
+      else
+        guard (Cstruct.length buf >= peer_info_start + 2) `Partial >>= fun () ->
```

Comment:
If we want to fully emulate the behavior of openvpn we should return `Ok None` in this case.

Reply:
Yes, correct. The question is if this is really sensible, but it is what openvpn does :-)